src/components/web3/api.ts

```diff
@@ -0,0 +1,309 @@
+      const thumbnail: string = nft.previews?.image_small_url
+        ? nft.previews.image_small_url
+        : nft.image_url;
+      /* TBD: uncomment this line, and delete the line following the comment...
+      result.push({ tooltip, thumbnail });
+*/
+      result.push(thumbnail);
+    });
+
+    return result;
+  } catch (error: any) {
+    console.error("!!! web3NFTs", error);
+    throw error;
+  }
+};
+
+export const web3NFTcollections = async (
+  address: string
+): Promise<NFTcollection[]> => {
+  const getNFTs = async (url: string): Promise<NFTcollection[]> => {
+    const response = await fetchWithTimeout(url, {
+      method: "GET",
+      headers: {
+        accept: "application/json",
+      },
+    });
+
+    const { status } = response;
+    if (status !== 200) {
+      throw new Error(`Request failed: ${status} ${response.statusText}`);
+    }
+
+    const page = await response.json();
+    let { nfts } = page;
+    const { next } = page;
+    if (next) {
+      nfts = nfts.concat(await getNFTs(next));
+    }
+
+    const collections = nfts.reduce(
+      (collections: { [key: string]: NFTcollection }, nft: NFT) => {
+        if (
+          nft?.collection?.collection_id &&
+          !collections[nft.collection.collection_id]
+        ) {
+          collections[nft.collection.collection_id] = {
+            id: nft.collection.collection_id,
+            name: nft.collection.name,
+            image_url: nft.image_url,
+          };
+        }
+
+        return collections;
+      },
+      {}
+    );
+
+    return Object.values(collections);
+  };
+
+  try {
+    const getNFTsByWalletURL = `${SIMPLEHASH_PROXY_ROOT_URL}/api/v0/nfts/owners?chains=ethereum&wallet_addresses=${address}`;
+    console.log(`>>> GET ${getNFTsByWalletURL}`);
+    return getNFTs(getNFTsByWalletURL);
+  } catch (error: any) {
+    console.error("!!! web3NFTcollections", error);
+    throw error;
+  }
+};
+
+const getNonce = async (): Promise<string> => {
+  const getNonceURL = "/api/v1/nonce";
+  const response = await fetchWithTimeout(getNonceURL, {
+    method: "GET",
+    headers: {
+      accept: "application/json",
+    },
+  });
+
+  const { nonce } = await response.json();
+  return nonce;
+};
+
+const STORAGE_KEY = "web3.authentication";
+
+export const web3Prove = async (
+  web3Address: string
+): Promise<Web3Authentication> => {
+  if (!web3Address) {
+    throw new Error("not logged into Web3");
+  }
+
+  window.web3 = new ethers.BrowserProvider(window.ethereum);
+  if (!window.web3) {
+    throw new Error("unable to create ethers.BrowserProvider object");
+  }
+
+  const nonce = await getNonce();
+  console.log("!!! nonce", nonce);
+  const message: EIP4361Message = {
+    domain: "talk.brave.com",
+    address: web3Address,
+    statement:
+      "Please sign this message so Brave Talk knows that you own this address",
+    uri: "https://talk.brave.com",
+    version: "1",
+    chainId: 1,
```

Comment:
I'm checking with the EIP-4361 authors what the expected behavior is here since we're not interacting with a chain here. It seems kind of odd to declare a chainId when the request is never being utilized by a chain.